core/src/main/scala/litter/CommutativeZeroSemigroup.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package litterbox
+
+import cats.kernel.CommutativeSemigroup
+
+import scala.{specialized => sp}
+
+/**
+ * CommutativeZeroSemigroup represents a commutative zero semigroup.
+ *
+ * A zero semigroup is commutative if for all x and y, x |+| y === y |+| x.
+ */
+trait CommutativeZeroSemigroup[@sp(Int, Long, Float, Double) A]
+    extends Any
+    with ZeroSemigroup[A]
+    with CommutativeSemigroup[A] { self =>
+  override def reverse: CommutativeZeroSemigroup[A] = self
+  override def intercalate(middle: A): CommutativeZeroSemigroup[A] =
```

Comment:
I wonder if we could have avoided this being necessary with a self type in Cats, but I don't think we can fix that.

Reply:
Agreed. Some other niceties in Cats would be extracting out `IntercalatedCommutativeSemigroup` and `ReversedSemigroup` classes that I could have extended here instead of copypasta.